dev/WindowsAppRuntime_BootstrapDLL/WindowsAppRuntime-Licensing.cpp

```diff
@@ -6,9 +6,125 @@
 
 #include <msixlicensing.h>
 
-STDAPI WindowsAppRuntime_InstallLicenses() noexcept try
+static std::filesystem::path GetPathForPackage(
+    PCWSTR packageFullName)
 {
-    RETURN_IF_FAILED(MsixInstallLicenses());
+    // Get the package's path
+    //
+    // NOTE: Use GetStagedPackagePathByFullName as don't care if the package is registered.
+    //
+    // NOTE: Don't use GetStagedPackagePathByFullName2 as we want the *install* location
+    //       (not e.g. modifiable location). Plus Get...2() is new in 20H1 (too new for us)
+    //       where Get...() exists down to Win81. In this case simplest is bestest.
+    wil::unique_cotaskmem_string packagePathBufferDynamic;
+    uint32_t packagePathLength{};
+    const auto rc{ GetStagedPackagePathByFullName(packageFullName, &packagePathLength, nullptr) };
+    if (rc != ERROR_INSUFFICIENT_BUFFER)
+    {
+        THROW_HR_MSG(HRESULT_FROM_WIN32(rc), "PackageFullName:%ls", packageFullName);
+    }
+    auto packagePath{ wil::make_cotaskmem_string(nullptr, packagePathLength) };
+    THROW_IF_WIN32_ERROR(GetStagedPackagePathByFullName(packageFullName, &packagePathLength, packagePath.get()));
+    std::filesystem::path path{ packagePath.get() };
+    return path;
+}
+
+static HMODULE LoadMsixLicensingApisAtPath(
+    const std::filesystem::path& path)
+{
+    auto windowsAppRuntimeDllFilename{ path };
+    windowsAppRuntimeDllFilename /= L"Microsoft.WindowsAppRuntime.dll";
+    auto windowsAppRuntimeDll(LoadLibraryEx(windowsAppRuntimeDllFilename.c_str(), nullptr, LOAD_WITH_ALTERED_SEARCH_PATH ));
+    if (!windowsAppRuntimeDll)
+    {
+        const auto lastError{ GetLastError() };
+        THROW_WIN32_MSG(lastError, "Error in LoadLibrary: %d (0x%X) loading %ls", lastError, lastError, windowsAppRuntimeDllFilename.c_str());
+    }
+    return windowsAppRuntimeDll;
+}
+
+static HMODULE LoadMsixLicensingApisInPackage(
+    PCWSTR packageFullName)
+{
+    auto path{ GetPathForPackage(packageFullName) };
+    return LoadMsixLicensingApisAtPath(path);
```

Comment:
Consider merging all these little temporaries together, like:

```c++
return LoadMsixLicensingApisAtPath(GetPathForPackage(packageFullName));
```

Reply:
Broken up this way is useful (easier) for debugging and same overhead in the end when the optimizer gets through with it (I would hope).